client/team_role_assignment.go

```diff
@@ -0,0 +1,82 @@
+package client
+
+type TeamRoleAssignmentPayload struct {
+	TeamId string `json:"teamId"`
+	Role   string `json:"role" tfschema:"role_id"`
+	Id     string `json:"id"`
+}
+
+type TeamRoleAssignmentCreateOrUpdatePayload struct {
+	TeamId         string `json:"teamId"`
+	Role           string `json:"role" tfschema:"role_id"`
+	EnvironmentId  string `json:"environmentId,omitempty"`
+	OrganizationId string `json:"organizationId,omitempty"`
+	ProjectId      string `json:"projectId,omitempty"`
```

Comment:
how can you add validation that 1 of them will be sent/exist?

Reply:
The terraform resource schema's forces correctness for the end-user.
If I add a validation (which I could) it would be used to detect developer bugs, and not end-user errors

I don't mind adding validations. However, it will not add much value (just a little bit).
Let me know - happy to add if you think it's required.

Reply:
> The terraform resource schema's forces correctness for the end-user.

Good enough for me